File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/news/domain/NewsScreen.java
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */

package pl.orbitemobile.wspolnoty.activities.news.domain;

import io.reactivex.Single;
import pl.orbitemobile.wspolnoty.data.RemoteRepository;
import pl.orbitemobile.wspolnoty.data.entities.Article;

public class NewsScreen {
    
    private RemoteRepository mRemoteRepository;
    
    public NewsScreen() {
        mRemoteRepository = new RemoteRepository();
    }
    
    public Single<Article[]> getRemoteArticles(int page) {
        return mRemoteRepository.getNews(page);
    }
    
}

File: app/src/main/java/pl/orbitemobile/wspolnoty/activities/news/domain/NewsScreen.kt
/*
 * Copyright (c) 2017. All Rights Reserved. Michal Jankowski orbitemobile.pl
 */

package pl.orbitemobile.wspolnoty.activities.news.domain

import io.reactivex.Single
import pl.orbitemobile.kotlinrx.fromIoToMainThread
import pl.orbitemobile.wspolnoty.data.RemoteRepositoryImpl
import pl.orbitemobile.wspolnoty.data.dto.ArticleDTO

class NewsScreen {

    private val mRemoteRepository: RemoteRepositoryImpl = RemoteRepositoryImpl.instance

    fun getRemoteArticles(page: Int): Single<Array<ArticleDTO>> =
            mRemoteRepository.getNews(page).fromIoToMainThread()

}
